src/Engine/Objects/Chest.cpp

```diff
@@ -89,23 +89,27 @@ bool Chest::open(int uChestID, int objectPid) {
                     pObjectZ = (pBLVFace->pBounding.z1 + pBLVFace->pBounding.z2) / 2;
                 }
             }
+
             dir_x = (double)pParty->vPosition.x - (double)pObjectX;
             dir_y = (double)pParty->vPosition.y - (double)pObjectY;
-            dir_z = ((double)pParty->sEyelevel + (double)pParty->vPosition.z) - (double)pObjectZ;
-            length_vector = sqrt((dir_x * dir_x) + (dir_y * dir_y) + (dir_z * dir_z));
+            length_vector = sqrt(dir_x * dir_x + dir_y * dir_y);
             if (length_vector <= 1.0) {
-                *(float*)&yawAngle = 0.0;
-                *(float*)&pitchAngle = 0.0;
+                yawAngle = 0;
+                pitchAngle = 0;
             } else {
-                pitchAngle = (int64_t)sqrt(dir_x * dir_x + dir_y * dir_y);
-                yawAngle = TrigLUT.atan2((int64_t) dir_x, (int64_t) dir_y);
-                pitchAngle = TrigLUT.atan2(dir_y * dir_y, (int64_t) dir_z);
+                // sprite should be rotated towards the party and moved slightly forward
+                yawAngle = TrigLUT.atan2((int64_t) dir_y, (int64_t) dir_x);
+                pitchAngle = 128;
             }
-            pDepth = 256;
-            if (length_vector < 256.0)
+
+            // offset distances so sprites appear in similar positions
+            constexpr int distances[4] = { 160, 100, 160, 32 };
```

Comment:
What's the logic behind this? X)

Reply:
Sprites are different sizes and have different centering options. I have revised PR to offset only in Z for consistency.

Reply:
Maybe it would be better to look into `SpriteObject::getSpriteFrame` and calculate offset based on sprite size?

Reply:
Good point, done it. Not sure how robust the math would be if we added other trap sprites.